### PR TITLE
Add support for `MintTokensTx` and `SetAccountDelegateTx`

### PR DESCRIPTION
**Pull request by jordipainan**

### CODE REVIEW

**Comment by jordipainan** [comment]
@mvdan can the failing test because of a false positive?, I think I need to import the `common` go-eth pkg

`vochain/state_test.go:8:2: package "github.com/ethereum/go-ethereum/common" is being imported more than once (ST1019)`

**Comment by p4u** [comment]
run `go mod tidy`

File: vochain/account.go [hunk 1]
```diff
@@ -282,3 +284,124 @@ func SetAccountInfoTxCheck(vtx *models.Tx, txBytes, signature []byte, state *Sta
...
+				return common.Address{}, common.Address{}, fmt.Errorf("delegate already added")
+			}
+		}
+		return sigAddress, delAcc, nil
+	case models.TxType_DEL_DELEGATE_FOR_ACCOUNT:
+		f := false
+		for i := 0; i < len(acc.DelegateAddrs); i++ {
+			delegateToCmp := common.BytesToAddress(acc.DelegateAddrs[i])
+			if delegateToCmp == delAcc {
+				f = true
```
Comment (p4u) [inline vochain/account.go]: just put the return here `return sigAddress, delAcc, nil` remove `f`and return by default `return common.Address{}, common.Address{}, fmt.Errorf("cannot remove a non existent delegate")`

File: vochain/state.go [hunk 2]
```diff
@@ -503,38 +503,63 @@ func (v *State) IsOracle(addr common.Address) (bool, error) {
 
 // setTreasurer saves the Treasurer address to the state
 func (v *State) setTreasurer(address common.Address) error {
+	t := &models.Treasurer{
+		Address: address.Bytes(),
+		Nonce:   0,
+	}
+	tBytes, err := proto.Marshal(t)
```
Comment (p4u) [inline vochain/state.go]:
```
tBytes, err := proto.Marshal(&models.Treasurer{
		Address: address.Bytes(),
		Nonce:   0,
	}
```

**Review by p4u — APPROVED** [review]

**Comment by p4u** [comment]
This is a good piece of code, congrats @jordipainan

**Pull request merged**